src/NHibernate.Test/NHSpecificTest/NH1914/Mappings.hbm.xml

```diff
@@ -1,54 +0,0 @@
-<?xml version="1.0" encoding="utf-8" ?>
-<hibernate-mapping xmlns="urn:nhibernate-mapping-2.2"
-									 assembly="NHibernate.Test"
-									 namespace="NHibernate.Test.NHSpecificTest.NH1914"
-									 default-lazy="false">
-
-	<class name="IDS" table="IDS">
-		<id name="Identifier">
-			<generator class="assigned" />
-		</id>
-		<property name="Name" not-null="true" />
-		<map name="CRSPLUTs" table="CRSPLUTs" cascade="all-delete-orphan">
-			<key column="ParentID"/>
-			<index column="OrganID" type="String"/>
-			<one-to-many class="ListOfHLUT"/>
-		</map>
-	</class>
-
-	<class name="ListOfHLUT" table="ListOfHLUT">
-		<id name="ID" column="CollectionID">
-			<generator class="native" />
-		</id>
-		<list name="Values" table="CustomValues" cascade="all">
-			<key column="ParentID"/>
-			<index column="Indexer" type="Int32"/>
-			<one-to-many class="HLUT"/>
-		</list>
-	</class>
-
-	<class name="HLUT" table="HLUT">
-
-		<!--NotWorking with below id generation.-->
-		<id name="Identifier" unsaved-value="0">
```

Comment:
Here was the actual bug of the user. Specifying that `0` is an unsaved value means that all other are to be considered as saved value, thus the behavior he was complaining about. Removing completely the `unsaved-value="0"` causes the test to succeed without the "fix", and that is the actual fix which should have been applied.

(Not specifying unsaved value with a assigned generator cause it to be handled as `undefined`, which causes NHibernate to ignore the id value for determining the entity state.)